### PR TITLE
[APPS-2784] Rollback from Java 21 to Java 17

### PR DESCRIPTION
**Pull request by SathishK-T**
https://hyland.atlassian.net/browse/APPS-2784

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.34.0
       - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.34.0
         with:
-          java-version: '21'
+          java-version: '17'
```
Comment (mstrankowski) [inline .github/workflows/ci.yml]: Wouldn't removing java version specified default to java 17?

File: samples/event-api-handlers/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM amazoncorretto:21-alpine3.18@sha256:e937a2668732ad0e138180f40e26586b4ba39282271889466ce6693410b77f2c
+FROM amazoncorretto:17-alpine3.18@sha256:e937a2668732ad0e138180f40e26586b4ba39282271889466ce6693410b77f2c
```
Comment (mstrankowski) [inline samples/event-api-handlers/Dockerfile]: Does the different java version have the same SHA? Seems weird, I don't think you can blindly replace this with a different number

File: samples/event-api-spring-integration/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM amazoncorretto:21-alpine3.18@sha256:e937a2668732ad0e138180f40e26586b4ba39282271889466ce6693410b77f2c
+FROM amazoncorretto:17-alpine3.18@sha256:e937a2668732ad0e138180f40e26586b4ba39282271889466ce6693410b77f2c
```
Comment (mstrankowski) [inline samples/event-api-spring-integration/Dockerfile]: same as above

File: samples/extension-template/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM amazoncorretto:21-alpine3.18@sha256:e937a2668732ad0e138180f40e26586b4ba39282271889466ce6693410b77f2c
+FROM amazoncorretto:17-alpine3.18@sha256:e937a2668732ad0e138180f40e26586b4ba39282271889466ce6693410b77f2c
```
Comment (mstrankowski) [inline samples/extension-template/Dockerfile]: same

File: alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml [hunk 5]
```diff
@@ -32,7 +32,7 @@
     <acs.search.api>${project.build.directory}/definitions/alfresco-search.yaml</acs.search.api>
     <acs.search-sql.api>${project.build.directory}/definitions/alfresco-search-sql.yaml</acs.search-sql.api>
 
-    <java.version>21</java.version>
+    <java.version>17</java.version>
     <maven.compiler.source>${java.version}</maven.compiler.source>
     <maven.compiler.target>${java.version}</maven.compiler.target>
```
Comment (mstrankowski) [inline alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml]: These versions shouldn't be needed here. There are versions specified in main pom and they seem to have been changed correctly. The properties here are redundant, as they should be inherited from the main pom.xml file from alfresco-java-sdk.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SathishK-T) [reply .github/workflows/ci.yml]: Yes, the default version is 17 hence removed java-version tag from here.

File: samples/event-api-handlers/Dockerfile (hunk 2 above)
Comment (SathishK-T) [reply samples/event-api-handlers/Dockerfile]: Yes updated the SHA.

File: samples/event-api-spring-integration/Dockerfile (hunk 3 above)
Comment (SathishK-T) [reply samples/event-api-spring-integration/Dockerfile]: Updated the SHA.

File: samples/extension-template/Dockerfile (hunk 4 above)
Comment (SathishK-T) [reply samples/extension-template/Dockerfile]: Updated the SHA.

File: alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml (hunk 5 above)
Comment (SathishK-T) [reply alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml]: Yes removed the redundant property.

Comment (mstrankowski) [reply alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml]: Remove also
    <maven.compiler.source>${java.version}</maven.compiler.source>
    <maven.compiler.target>${java.version}</maven.compiler.target>
    
Those are defined in main pom.xml too.

Comment (SathishK-T) [reply alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml]: Removed the below property from - alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib/pom.xml
<maven.compiler.source>${java.version}</maven.compiler.source>
<maven.compiler.target>${java.version}</maven.compiler.target>

**Review by mstrankowski — APPROVED** [review]

**Pull request merged**